enforcer/datapath_tcp.go

```
@@ -946,22 +928,13 @@ func (d *Datapath) appRetrieveState(p *packet.Packet) (*PUContext, *TCPConnectio
 func (d *Datapath) netSynRetrieveState(p *packet.Packet) (*PUContext, *TCPConnection, error) {
 
 	context, err := d.contextFromIP(false, p.DestinationAddress.String(), p.Mark, strconv.Itoa(int(p.DestinationPort)))
-
+	// if PU context could be not obtained from the IP for the Syn packet,
+	// try to fetch it from packet Mark
 	if err != nil {
-		//This needs to hit only for local processes never for containers
-		//Don't return an error create a dummy context and return it so we truncate the packet before we send it up
-		if d.mode != constants.RemoteContainer {
-
-			context = &PUContext{
-				PUType: constants.TransientPU,
-			}
-			//we will create the bare minimum needed to exercise our stack
-			//We need this syn to look similar to what we will pass on the retry
-			//so we setup enought for us to identify this request in the later stages
-			return context, nil, nil
+		context, err = d.contextFromMark(p.DestinationAddress.String(), p.Mark, strconv.Itoa(int(p.DestinationPort)))
+		if err != nil {
+			return nil, nil, nil
```

You should return the error here